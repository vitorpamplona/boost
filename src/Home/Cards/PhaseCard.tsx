import React, { FunctionComponent, useEffect, useState } from "react"
import { View , StyleSheet, Text,TouchableOpacity} from "react-native"


import {
    Colors,
    Typography,
    Affordances,
  } from "../../styles"
import {getPhase, getPhaseMessage, removePhase,setLocalPhase, setPhaseMessage } from "../../utils/storage"
import Pen from "../../assets/svgs/pen"
 
  


const Options: FunctionComponent<{text:string,func:Function}> = ({text,func})=>{

    return(
         <TouchableOpacity   style={styles.optioncontainer} onPress={func}>
             <Text>{text}</Text>
         </TouchableOpacity>
    );
}

const PhaseCard: FunctionComponent  = ()=>{
  const [Phase, setPhase]= useState<string|null>("random words to prevent it from rendering");
  const [Message, setMessage]= useState<string|null>(null);

  const setPhaseLocal=(item:string,message:string)=>{
     setLocalPhase(item);
     setPhaseMessage(message)
     setPhase(item)
     setMessage(message)

  }
  const getPhaseData=()=>{
    getPhase().then((value)=>{
      value?setPhase(value):setPhase(null) 
    })
    getPhaseMessage().then((message)=>{
      message?setMessage(message):setMessage(null)
    })
  }

  useEffect(()=>{
    getPhaseData()
  },[Phase])

  const editClicked =()=>{
    removePhase().then(()=>{
      setPhase(null);
    })
  }

  if(!Phase)
  {    
  return(
        <View style={styles.container}>
          <Text style={styles.header}>Choose whichever is applicable</Text>
            <Options text="I am a Frontline essential worker" func={()=>{setPhaseLocal("1b","I am a Frontline essential worker")}} />
            <Options text=" I am aged 75 years and older" func={()=>{setPhaseLocal("1b","I am aged 75 years and older")}} />
            <Options text="I am 65—74 years old" func={()=>{setPhaseLocal("1c","I am 65—74 years old")}} />
            <Options text="I am 16—64 years with underlying medical conditions" func={()=>{setPhaseLocal("1c","I am 16—64 years with underlying medical conditions")}}  /> 
            <Options text="None of the above"  func={()=>{setPhaseLocal("normal","None of the above")}}  /> 
             </View> 
        
    );
  }
  else{
    return <View style={styles.container}>
      <View style={{flexDirection:"row", justifyContent:"space-between"}}>
       <Text style={styles.headerVacc}>your vaccination Phase: {Phase}</Text>
       <TouchableOpacity onPress={()=>{editClicked()}}>
       <Pen/>
       </TouchableOpacity>

      </View>

    <View>
      <Text style={styles.headerVacc}>you choose: {Message}</Text>
    </View>
       </View>;
  }
}

   

const styles= StyleSheet.create({
    container:{
        ...Affordances.floatingContainer,
        
    },
    header:{
            ...Typography.header.x40,
            color: Colors.neutral.black,
    },
    headerVacc:{
      ...Typography.header.x20,
      color: Colors.neutral.black,
    },
  optioncontainer:{
    padding:10,
    marginVertical:10,
    backgroundColor: "#C186FC",
    borderRadius:10,
    alignItems:"center"


  }
})
export default PhaseCard;
