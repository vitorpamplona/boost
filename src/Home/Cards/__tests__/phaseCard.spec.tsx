
import React from 'react';
import {create} from 'react-test-renderer';
import PhaseCard from '../PhaseCard';

const tree = create(<PhaseCard/>)
test('renders correctly', () => { 
    expect(tree).toMatchSnapshot();
  });