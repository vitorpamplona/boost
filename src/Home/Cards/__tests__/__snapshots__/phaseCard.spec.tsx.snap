// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 16,
      "elevation": 20,
      "marginBottom": 24,
      "paddingHorizontal": 24,
      "paddingVertical": 24,
      "shadowColor": "#3c475b",
      "shadowOffset": Object {
        "height": 8,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 13.16,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "IBMPlexSans-SemiBold",
          "fontSize": 16,
          "fontWeight": "600",
          "letterSpacing": 0.25,
          "lineHeight": 24,
        }
      }
    >
      your vaccination Phase: 
      random words to prevent it from rendering
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Svg
        height={25}
        viewBox="0 0 18 18"
        width={25}
      >
        <G
          fill="none"
          fillRule="evenodd"
        >
          <Path
            d="M-3-3h24v24H-3z"
          />
          <Path
            d="M0 14.46v3.04c0 .28.22.5.5.5h3.04c.13 0 .26-.05.35-.15L14.81 6.94l-3.75-3.75L.15 14.1c-.1.1-.15.22-.15.36zM17.71 4.04a.996.996 0 000-1.41L15.37.29a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            fill="#1D1D1D"
          />
        </G>
      </Svg>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "IBMPlexSans-SemiBold",
          "fontSize": 16,
          "fontWeight": "600",
          "letterSpacing": 0.25,
          "lineHeight": 24,
        }
      }
    >
      you choose: 
    </Text>
  </View>
</View>
`;
